### PR TITLE
Feature/generate recovery kit as pdf

### PR DESCRIPTION
**Pull request by aince42**

### CODE REVIEW

File: packages/identity_recovery_kit/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<!--
```
Comment (nicole-eb) [inline packages/identity_recovery_kit/README.md:1]: I think you could shortly summarize what this package is, like the readme of the other packages

File: packages/identity_recovery_kit/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  #       - asset: fonts/Schyler-Italic.ttf
+  #         style: italic
+  #   - family: Trajan Pro
+  #     fonts:
+  #       - asset: fonts/TrajanPro.ttf
+  #       - asset: fonts/TrajanPro_Bold.ttf
+  #         weight: 700
+  #
+  # For details regarding fonts in packages, see
+  # https://flutter.dev/to/font-from-package
```
Comment (nicole-eb) [inline packages/identity_recovery_kit/pubspec.yaml:56]: Can be removed

File: packages/identity_recovery_kit/LICENSE [hunk 3]
```diff
@@ -0,0 +1 @@
+TODO: Add your license here.
```
Comment (nicole-eb) [inline packages/identity_recovery_kit/LICENSE:1]: Needs to be updated

**Review by nicole-eb — CHANGES_REQUESTED** [review]

File: packages/identity_recovery_kit/CHANGELOG.md
Comment (jkoenig134) [inline packages/identity_recovery_kit/CHANGELOG.md:1]: delete this file.

File: packages/identity_recovery_kit/analysis_options.yaml
Comment (jkoenig134) [inline packages/identity_recovery_kit/analysis_options.yaml:1]: see other packages and examples on how to setup lints.

File: packages/identity_recovery_kit/example/README.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+This project is a starting point for a Flutter application.
+
+A few resources to get you started if this is your first Flutter project:
+
+- [Lab: Write your first Flutter app](https://docs.flutter.dev/get-started/codelab)
+- [Cookbook: Useful Flutter samples](https://docs.flutter.dev/cookbook)
+
+For help getting started with Flutter development, view the
+[online documentation](https://docs.flutter.dev/), which offers tutorials,
+samples, guidance on mobile development, and a full API reference.
```
Comment (jkoenig134) [inline packages/identity_recovery_kit/example/README.md:16]: why all the boilerplate stuff, please don't check in everything recklessly.

File: packages/identity_recovery_kit/example/analysis_options.yaml
Comment (jkoenig134) [inline packages/identity_recovery_kit/example/analysis_options.yaml:1]: wrong lint file.

File: patches.md
Comment (jkoenig134) [inline patches.md:1]: OMG, check what you commit.

File: packages/identity_recovery_kit/example/pubspec.yaml [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  #       - asset: fonts/Schyler-Italic.ttf
+  #         style: italic
+  #   - family: Trajan Pro
+  #     fonts:
+  #       - asset: fonts/TrajanPro.ttf
+  #       - asset: fonts/TrajanPro_Bold.ttf
+  #         weight: 700
+  #
+  # For details regarding fonts from package dependencies,
+  # see https://flutter.dev/to/font-from-package
```
Comment (jkoenig134) [inline packages/identity_recovery_kit/example/pubspec.yaml:90]: remove all the comments, also above.

File: packages/identity_recovery_kit/example/lib/main.dart [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+      body: Padding(
+        padding: const EdgeInsets.all(16),
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          crossAxisAlignment: CrossAxisAlignment.center,
+          children: [
+            TextField(
+              controller: _controller,
+              decoration: const InputDecoration(border: OutlineInputBorder(), labelText: 'QR data'),
+            ),
```
Comment (jkoenig134) [inline packages/identity_recovery_kit/example/lib/main.dart:80]: we talked about having way more stuff to setup (headerTitle, ...) in general **everything** that can be configured should be able to configure.

File: packages/identity_recovery_kit/example/lib/main.dart [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+const headerTitle = 'Wiederherstellungskit für\nIhre Mein Bildungsraum-Daten';
+const keepSafeText =
+    'Bitte bewahren Sie dieses Wiederherstellungs-Kit an einem für Dritte unzugänglichen Ort auf. Sobald Sie das selbst gewählte Master Passwort hier notiert haben ist ein Zugriff auf Ihre Mein Bildungsraum-Daten möglich.';
+const infoText1 =
+    'Drucken Sie dieses Dokument aus oder speichern Sie es an einem sicheren Ort, beispielsweise einem weiteren Mobilgerät oder Ihrem Computer.';
+const infoText2 =
+    'Schreiben Sie in das untenstehende Feld das Passwort, das Sie bei der Erstellung des Kits vergeben haben. Zur Wiederherstellung Ihrer verschlüsselten Daten benötigen Sie den QR Code und das von Ihnen vergebene Passwort.';
+const idLabel = 'Mein Bildungsraum ID';
+const id = 'did:e:nmshd-bkb.meinbildungsraum.de:dids:4fe203de633a66b56730e3';
```
Comment (jkoenig134) [inline packages/identity_recovery_kit/example/lib/main.dart:48]: why do you call this ID, when this is **obviously** an address?

File: packages/identity_recovery_kit/example/lib/main.dart [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+const passwordLabel = 'Passwort';
+const qrDescription =
+    'Wenn Sie Ihr Endgerät verloren haben sollten oder keinen Zugriff mehr darauf haben, scannen Sie einfach den QR Code mit einem neuen Endgerät um wieder Zugang zu Ihren Mein Bildungsraum-Daten zu erhalten.';
+const needHelpTitle = 'Benötigen Sie Hilfe?';
+const needHelpText = 'Wenden Sie sich an den Mein Bildungsraum-Support: support@meinbildungsraum.de';
```
Comment (jkoenig134) [inline packages/identity_recovery_kit/example/lib/main.dart:53]: hardcoded stuff has to be english. And pls remove Bildungsraum mentions

File: packages/identity_recovery_kit/lib/src/pdf_generator.dart [hunk 9]
```diff
@@ -0,0 +1,261 @@
...
+        }
+      }
+    }
+
+    final picture = pictureRecorder.endRecording();
+    final image = await picture.toImage(size, size);
+
+    final byteData = await image.toByteData(format: ui.ImageByteFormat.png);
+    return byteData!.buffer.asUint8List();
+  }
```
Comment (jkoenig134) [inline packages/identity_recovery_kit/lib/src/pdf_generator.dart:260]: why isn't this in another file?

File: packages/identity_recovery_kit/lib/src/pdf_generator.dart [hunk 10]
```diff
@@ -0,0 +1,261 @@
...
+    required String borderHexColor,
+    required String labelHexColor,
+    required String idHexColor,
+  }) async {
+    final pdf = pw.Document();
+
+    final logoImage = pw.MemoryImage(logoBytes);
+
+    final qrCodeBytes = await _generateQrCode(truncatedReference);
+    final qrImage = pw.MemoryImage(qrCodeBytes);
```
Comment (jkoenig134) [inline packages/identity_recovery_kit/lib/src/pdf_generator.dart:35]: why does `_generateQrCode` not return a `pw.MemoryImage`?

File: TODO.txt
Comment (jkoenig134) [inline TODO.txt:1]: ???

**Review by jkoenig134 — CHANGES_REQUESTED** [review]

File: packages/identity_recovery_kit/README.md [hunk 11]
```diff
@@ -0,0 +1,11 @@
+# identity recovery kit
```
Comment (jkoenig134) [inline packages/identity_recovery_kit/README.md:1]:
```suggestion
# Identity Recovery Kit Generator
```

**Review by jkoenig134 — APPROVED** [review]

**Pull request merged**